Add tutorials

Abdusshh opened the pull request.

mdogan commented:
Can you add a summary and/or title of the added tutorials to the commit message and PR description?

Abdusshh commented:
Sorry about that. I added the following tutorials:

Vector
Create and Deploy RAG Applications with Gradio (vector/tutorials/gradio-application.mdx)

Redis
Rate Limiting for Your FastAPI App (redis/tutorials/python_rate_limiting.mdx)
Multithreaded Web Scraping with Redis Caching (redis/tutorials/python_multithreading.mdx)

I also modified redis/tutorials/python_url_shortener.mdx and redis/tutorials/python_fastapi_caching.mdx to correct the environment variable names.
